src/ensembl/src/content/app/entity-viewer/gene-view/components/unspliced-transcript/UnsplicedTranscript.tsx

```diff
@@ -199,12 +194,12 @@ const ExonBlock = (props: ExonBlockProps) => {
   const y = -3;
   const height = BLOCK_HEIGHT;
 
-  const exonClasses = classNames(styles.exon, props.className);
+  const exonClasses = props.className;
 
   if (cds && isNonCodingLeft) {
     const nonCodingPart = (
       <rect
-        className={classNames(exonClasses, styles.emptyBlock)}
+        className={classNames(exonClasses, styles.emptyBlock) || undefined}
```

Comment:
Think `classNames` will return `undefined` if the classes passed to it are not available. So don't think we need `||` to assign `undefined`.

Reply:
Try it 😄 That's not what the [type definition](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/classnames/types.d.ts#L11) says

Reply:
@azangru Thanks, I checked it out and was surprised it sends back an empty string rather than `undefined`. Do we still need the `undefined` in there? Just curious about the usage of it here as we haven't used it in many other places.

Reply:
If we pass an empty string to the `className` attribute of a jsx element, it will render an empty class attribute in the DOM:

![image](https://user-images.githubusercontent.com/6834224/110155715-eea57f80-7ddd-11eb-9e53-8b482b8c2018.png)

If we pass `undefined` to the `className` attribute, the rendered DOM element will note have a class attribute at all:

![image](https://user-images.githubusercontent.com/6834224/110155847-20b6e180-7dde-11eb-8ff5-0e95824dc0df.png)

I think the second option is much more preferable.

Reply:
> Just curious about the usage of it here as we haven't used it in many other places.

I think we are fairly sure in most other places that the value that we are passing to `className` will not be an empty string. This component turned out to be a rare exception.

Reply:
@azangru Thanks for the explanation. It makes a lot more sense now.